lib/kubernetes-deploy/statsd.rb

```diff
@@ -22,5 +22,30 @@ def self.build
       end
       ::StatsD.backend
     end
+
+    module MeasureMethods
+      def measure_method(method_name, metric = nil)
+        unless method_defined?(method_name) || private_method_defined?(method_name)
+          raise NotImplementedError, "Cannot instrument undefined method #{method_name}"
+        end
+
+        unless const_defined?("InstrumentationProxy")
+          const_set("InstrumentationProxy", Module.new)
+          should_prepend = true
+        end
+        instrumentation_proxy = const_get("InstrumentationProxy")
+        metric ||= "#{method_name}.duration"
+
+        instrumentation_proxy.send(:define_method, method_name) do |*args, &block|
+          start_time = Time.now.utc
+          result = super(*args, &block)
```

Comment:
If the `super` raises an error then you will never send the metric. Is that on purpose? Maybe you want to send it with a `error: true` tag?

Reply:
Not on purpose. Added that, with tests.